crates/weaver_resolved_schema/src/lib.rs

```diff
@@ -92,23 +146,375 @@ impl ResolvedTelemetrySchema {
+                        SchemaItemType::Attributes,
+                        SchemaItemChange::Added {
+                            name: attr.name.clone(),
+                        },
+                    );
+                }
+            }
+        }
+
+        // Any attribute in the baseline schema that is not present in the latest schema
+        // is considered removed.
+        // Note: This should never occur if the registry evolution process is followed.
+        // However, detecting this case is useful for identifying a violation of the process.
+        for (attr_name, attr) in baseline_attributes.iter() {
+            if !latest_attributes.contains_key(attr_name) {
+                changes.add_change(
+                    SchemaItemType::Attributes,
+                    SchemaItemChange::Removed {
+                        name: attr.name.clone(),
+                    },
+                );
+            }
+        }
+
+        // The attribute names that remain in the list `renamed_attributes` are those
+        // present in both versions of the same schema. They represent cases where
+        // attributes have been renamed to an already existing attribute.
+        for (new_name, old_names) in renamed_attributes.iter() {
+            changes.add_change(
+                SchemaItemType::Attributes,
+                SchemaItemChange::RenamedToExisting {
+                    old_names: old_names.iter().map(|n| (*n).to_owned()).collect(),
+                    current_name: (*new_name).to_owned(),
+                },
+            );
+        }
+    }
+
+    fn diff_signals(
+        &self,
+        schema_item_type: SchemaItemType,
+        latest_signals: &HashMap<String, &Group>,
+        baseline_signals: &HashMap<String, &Group>,
+        changes: &mut SchemaChanges,
+    ) {
+        /// Get the name of the provided group based on the given schema item type.
+        fn group_name(schema_item_type: SchemaItemType, group: &Group) -> String {
+            match schema_item_type {
+                SchemaItemType::Attributes
+                | SchemaItemType::Events
+                | SchemaItemType::Spans
+                | SchemaItemType::Resources => group.name.clone().unwrap_or_default(),
+                SchemaItemType::Metrics => group.metric_name.clone().unwrap_or_default(),
+            }
+        }
+
+        // A map of signal groups that have been renamed to a new signal of same type.
+        // The key is the new name of the signal, and the value is a set of old names.
+        // The key may refer to an existing signal in the baseline schema or
+        // a new signal in the latest schema.
+        let mut renamed_signals: HashMap<String, Vec<&Group>> = HashMap::new();
+
+        // Collect all the information related to the signals that have been
+        // deprecated in the latest schema.
+        for (signal_name, group) in latest_signals.iter() {
+            if let Some(deprecated) = group.deprecated.as_ref() {
```

Comment:
Again, I think a comparison with the baseline is missing.
```suggestion
            if let Some(deprecated) = group.deprecated.as_ref() {
                // is this a change from the baseline?
                if let Some(baseline_group) = baseline_signals.get(signal_name) {
                    if let Some(baseline_deprecated) = baseline_group.deprecated.as_ref() {
                        if deprecated == baseline_deprecated {
                            continue;
                        }
                    }
                }
```

Reply:
Good catch! Thank you

Reply:
Updated an existing test to catch this issue.